llvm/lib/Target/AIE/AIE2Subtarget.cpp

```diff
@@ -76,3 +76,53 @@ const RegisterBankInfo *AIE2Subtarget::getRegBankInfo() const {
 InstructionSelector *AIE2Subtarget::getInstructionSelector() const {
   return InstSelector.get();
 }
+
+std::bitset<32>
+AIE2Subtarget::getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
+  using namespace AIE2;
+  std::bitset<32> MemoryBanks;
+
+  switch (static_cast<AddressSpaces>(AddrSpace)) {
+  case AddressSpaces::a:
+    MemoryBanks.set(static_cast<int>(DMBanks::A));
+    break;
+  case AddressSpaces::b:
+    MemoryBanks.set(static_cast<int>(DMBanks::B));
```

Comment:
I recommend to cast to `unsigned` here, because the parameter is `std::size_t` and we have the same definition as in `C`:

> C Standard Section 4.1.5 defines size_t as an unsigned integral type of the result of the sizeof operator